creepy/subprocess/pypen.py

```diff
@@ -108,6 +131,25 @@ def detach(self):
             del self._send, self._recv
         except AttributeError:
             pass
+        try:
+            os.remove(self._out_path)
+            os.remove(self._in_path)
+        except (OSError, FileNotFoundError, AttributeError):
+            pass
+
+
+def _make_fifo(path: str | None = None):
+    """Open FIFO at specified path or create and open a new one."""
+    if path is None:
+        path = os.path.join('/tmp', secrets.token_urlsafe(16))
+        os.mkfifo(path)
+    # Open the FIFO for reading. `os.O_NONBLOCK` is used in order to prevent blocking the process when opening.
+    in_fd = os.open(path, os.O_RDONLY | os.O_NONBLOCK)
+    # Since we already opened the FIFO for reading, following open won't block.
+    out_fd = os.open(path, os.O_WRONLY)
+    # Set the file descriptor to blocking mode in order to synchronize the execution of the two processes.
```

Comment:
ChatGPT doesn't agree with my comment:
> When you use `fcntl.fcntl(in_fd, fcntl.F_SETFL, os.O_RDONLY)`, you're not just removing the `os.O_NONBLOCK` flag; you're actually setting the file descriptor's flags to `os.O_RDONLY` exclusively. This action overrides all existing flags, potentially leading to unintended consequences.

However, there is no such operation as "remove flag" in `fcntl`. It proposes following approach:
```py
# Retrieve the current flags
current_flags = fcntl.fcntl(in_fd, fcntl.F_GETFL)

# Clear the os.O_NONBLOCK flag
new_flags = current_flags & ~os.O_NONBLOCK

# Apply the updated flags
fcntl.fcntl(in_fd, fcntl.F_SETFL, new_flags)
```
As far as I see it is the same logic as I do in my function and it's not hard to understand with new comments. If I'm missing something or a clarification is still needed please let me know.